From bdb383d7aa6877761c2ff7707ed8889a7f692c88 Mon Sep 17 00:00:00 2001
From: Camelia Groza <camelia.groza@nxp.com>
Date: Tue, 19 Jun 2018 14:23:28 +0800
Subject: [PATCH 551/767] sdk_dpaa: ceetm: stop the netdev queues when
 switching ceetm on

[context adjustment]

Signed-off-by: Camelia Groza <camelia.groza@nxp.com>
Integrated-by: Zhao Qiang <qiang.zhao@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_ceetm.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_ceetm.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_ceetm.c
index 45568c154f7b..2515e8159e2e 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_ceetm.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_ceetm.c
@@ -1098,7 +1098,9 @@ static int ceetm_init(struct Qdisc *sch, struct nlattr *opt)
 
 	switch (priv->type) {
 	case CEETM_ROOT:
+		netif_tx_stop_all_queues(dev);
 		ret = ceetm_init_root(sch, priv, qopt);
+		netif_tx_wake_all_queues(dev);
 		break;
 	case CEETM_PRIO:
 		ret = ceetm_init_prio(sch, priv, qopt);
-- 
2.17.0

